### ISSUE TITLE
Error boundary for compositions

### ISSUE THREAD

**Issue by KutnerUri** [teambit/bit#3791]
When trying to render a faulty component, for example:

```tsx
// button.compositions.tsx
export const Problematic = "this cannot render";
```

React may fail, and only show errors in the console:
![Screen Shot 2021-02-15 at 15 03 46](https://user-images.githubusercontent.com/5400361/107952961-1fdf1c80-6fa3-11eb-9611-87f7f1d1f3e4.jpg)

This is commonly solved with **React Error Boundary** introduced in React 16.
Using the very popular `react-error-boundary` package, errors are shown using a fallback:
![Screen Shot 2021-02-15 at 15 26 50](https://user-images.githubusercontent.com/5400361/107952995-27062a80-6fa3-11eb-90dc-885f3f32b4f8.jpg)

We can style the fallback however we want, and we will need to add this both for compositions (full page view), and docs (card view).

## React Error Overlay
React Error Boundary is **NOT** a substitute for React's Error Overlay.
As can be seen [here](https://github.com/facebook/create-react-app/issues/6530), Error Overlay should **still show up, even if Error Boundary catches** the error.

I'm not sure why our custom Error Overlay doesn't catch render errors, I will try to investigate.

**Comment by itaymendel** [comment]
Agreed on a potential solution for it.

**Comment by ranm8** [comment]
@KutnerUri @itaymendel do we have a product spec defined for this task? Any thoughts re design here? I think this makes sense.

**Comment by KutnerUri** [comment]
closed by #4400
